src/test/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/compute/v5_4/OpenNebulaComputePluginTest.java

```diff
@@ -97,761 +97,441 @@
 
 		// exercise
-		this.plugin.updateHardwareRequirements(client);
+		this.plugin.doRequestInstance(this.client, createComputeRequest);
 
 		// verify
-		PowerMockito.verifyStatic(OpenNebulaClientUtil.class, VerificationModeFactory.times(1));
-		OpenNebulaClientUtil.getImagePool(Mockito.any(Client.class));
-
-		PowerMockito.verifyStatic(OpenNebulaClientUtil.class, VerificationModeFactory.times(1));
-		OpenNebulaClientUtil.getTemplatePool(Mockito.any(Client.class));
+		Mockito.verify(createComputeRequest, Mockito.times(TestUtils.RUN_ONCE)).getVirtualMachine();
 
-		Mockito.verify(image, Mockito.times(1)).xpath(Mockito.anyString());
-		Mockito.verify(template, Mockito.times(1)).getId();
-		Mockito.verify(template, Mockito.times(1)).getName();
-		Mockito.verify(template, Mockito.times(3)).xpath(Mockito.anyString());
-		Mockito.verify(this.plugin, Mockito.times(1)).getDiskSizeFromImageSizeMap(Mockito.anyMap(), Mockito.anyString());
+		PowerMockito.verifyStatic(OpenNebulaClientUtil.class, Mockito.times(TestUtils.RUN_ONCE));
+		OpenNebulaClientUtil.allocateVirtualMachine(this.client, createComputeRequest.getVirtualMachine().marshalTemplate());
 	}
-    
-	// test case: When calling the getComputeInstance method passing a valid virtual
-	// machine, it must obtain the data to mount an instance of this resource.
+
+	// test case: when invoking listNetworkIds with a non-empty list of network ids,
+    // a new network ids list should be returned containing all of the passed list ids
+	// and the default network id.
 	@Test
-	public void testGetComputeInstanceSuccessfully() {
+	public void testListNetworkIds() {
 		// set up
-		String id = FAKE_ID;
-		String name = FAKE_NAME;
-		int cpu = CPU_VALUE_1;
-		int memory = MEMORY_VALUE_1024;
-		int disk = DISK_VALUE_8GB;
-		List<String> ipAddresses = new ArrayList<String>();
-		ipAddresses.add(IP_ADDRESS_ONE);
-		ipAddresses.add(IP_ADDRESS_TWO);
-
-		VirtualMachine virtualMachine = Mockito.mock(VirtualMachine.class);
-		Mockito.when(virtualMachine.getId()).thenReturn(id);
-		Mockito.when(virtualMachine.getName()).thenReturn(name);
-		Mockito.when(virtualMachine.lcmStateStr()).thenReturn(LCM_STATE_RUNNING);
-		Mockito.when(virtualMachine.xpath(OpenNebulaComputePlugin.TEMPLATE_CPU_PATH)).thenReturn(ONE_STRING_VALUE);
-		Mockito.when(virtualMachine.xpath(OpenNebulaComputePlugin.TEMPLATE_DISK_SIZE_PATH))
-				.thenReturn(IMAGE_SIZE_STRING_VALUE);
-		Mockito.when(virtualMachine.xpath(OpenNebulaComputePlugin.TEMPLATE_MEMORY_PATH))
-				.thenReturn(MEMORY_STRING_VALUE);
-
-		String xml = getVirtualMachineResponse();
-		OneResponse response = Mockito.mock(OneResponse.class);
-		Mockito.when(virtualMachine.info()).thenReturn(response);
-		Mockito.when(response.isError()).thenReturn(false);
-		Mockito.when(response.getMessage()).thenReturn(xml);
-
-		ComputeInstance expected = new ComputeInstance(id, OpenNebulaStateMapper.COMPUTE_RUNNING_STATE, name, cpu, memory, disk, ipAddresses);
+		String defaultNetworkId = "0";
```

Comment:
Use a constant to set this value of this variable.
Do this for other occurrences ... Including, if possible, some checks to avoid magic numbers.

Reply:
done